datafusion/expr/src/udf.rs

```diff
@@ -345,12 +351,22 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
     fn name(&self) -> &str;
 
     /// Returns the user-defined display name of the UDF given the arguments
-    ///
     fn display_name(&self, args: &[Expr]) -> Result<String> {
-        let names: Vec<String> = args.iter().map(create_name).collect::<Result<_>>()?;
+        let names: Vec<String> = args.iter().map(ToString::to_string).collect();
+        // TODO: join with ", " to standardize the formatting of Vec<Expr>, <https://github.com/apache/datafusion/issues/10364>
         Ok(format!("{}({})", self.name(), names.join(",")))
     }
 
+    /// Returns the user-defined schema name of the UDF given the arguments
```

Comment:
```suggestion
    /// Returns the name of the column this expression would create
    ///
    /// See [`Expr::schema_name`] for details
```

Maybe it is also worth doing the `impl Display` thing here too which would allow avoiding these string allocations

Reply:
It seems I need another `SchemaFunctionDisplay` for replacing String with impl Display in UDFImpl. I would like to hold on to the change for function

Reply:
makes sense -- I don't think there is any reason that this impl Display thing has to be done -- I was thinking it might be worth exploring if we were already changing the code anyways